C# utlity classes for export. (Bottom-up)

mr-phazer opened the pull request:
Add; 
- `PackedMeshBuilder `updated to be more S.O.L.I.D, it is now "open closed". Its interface `IPackedMeshBuilder `has two implementations, `IndexedPackedMeshBuilder`, and `WeightedIndexedPackedMeshBuilder`.  the scene builder uses
`PackedMeshBuilderFactory.GetBuilder(AnimationFile skeletonFile)` to obtain the riggedd of non-rigged version as needed.

- `VertexWeightCreator` is used  to copy the 1-4 weights in an RMV2 vertex to 1 un-nested array, 

- Added  `Animations.GetIdFromBoneName(string name)` for save lookups (not out of range),  AnimationFile already has `string GetBoneNameFromId(int id)

- 2-3 minor helper classes

donkeyProgramming commented on CommonControls/FileTypes/Animation/AnimationFile.cs:
```diff
@@ -181,10 +181,18 @@ public int GetIdFromBoneName(string name)
+        /// <summary>
+        /// Safe method for looking up bones, as opposed to using the bone index directly
+        /// </summary>
+        public string GetBoneNameFromIndex(int boneIndex)
```
this should not be added here. 
Feel free to add it as an extention method within the assetManagment project or as a helper function where used

mr-phazer replied:
Yeah, oops,, I get it :)
I will remove both
`string GetBoneNameFromIndex(int boneIndex)`
`public int GetIdFromBoneName(string name)` 
from` AnimationFIle`
to `class AsseetMagement.AnimationProcessing.SkeletonHelper`

donkeyProgramming reviewed (changes requested):
Unless absolutly required, dont modify files outside of the assetmanagementproject

mr-phazer commented:
Changes done:
- Moved:
  -`string GetBoneNameFromIndex(int boneIndex)`
  -`int GetIdFromBoneName(string name)`
from` class AnimationFIle`
to class `AsseetMagement.AnimationProcessing.SkeletonHelper`
- Changed reference to the methods in `AsseetMagement` to the new methods in `SkeletonHelper`
